server/src/main/java/org/elasticsearch/index/engine/Engine.java

```diff
@@ -759,10 +763,27 @@ private void fillSegmentInfo(SegmentReader segmentReader, boolean search, Map<St
         }
         segment.segmentSort = info.info.getIndexSort();
         segment.attributes = info.info.getAttributes();
+        segment.docsWithSource = docsWithSource(segmentReader, info);
         // TODO: add more fine grained mem stats values to per segment info here
         segments.put(info.info.name, segment);
     }
 
+    private int docsWithSource(SegmentReader reader, SegmentCommitInfo info) {
```

Comment:
I'm still calling this `docsWithSource` internally, as that's what we're actually retrieving.  The external name is now `fully_merged_docs`.